modules/git/repo.go

```diff
@@ -146,8 +149,22 @@ func CloneWithArgs(ctx context.Context, from, to string, args []string, opts Clo
 		opts.Timeout = -1
 	}
 
-	_, err = cmd.RunTimeout(opts.Timeout)
-	return err
+	var envs []string
+	u, err := url.Parse(from)
+	if err == nil && (strings.EqualFold(u.Scheme, "http") || strings.EqualFold(u.Scheme, "https")) {
+		if proxy.Match(u.Host) {
+			envs = []string{fmt.Sprintf("https_proxy=%s", proxy.GetProxyURL())}
+		}
+	}
+
+	var stderr = new(bytes.Buffer)
+	err = cmd.RunWithContext(&RunContext{
+		Timeout: opts.Timeout,
+		Env:     envs,
+		Stdout:  new(bytes.Buffer),
```

Comment:
If we're not going to use the output can't we just set it to io.Discard?

```suggestion
		Stdout:  io.Discard,
```